apps/dashboard/src/app/(main)/(protected)/projects/[projectId]/project-settings/page-client.tsx

```diff
@@ -17,6 +17,33 @@ const projectInformationSchema = yup.object().shape({
   description: yup.string(),
 });
 
+// Memoized team member list item component
+const TeamMemberItem = React.memo(function TeamMemberItem({ member }: { member: any }) {
```

Comment:
why do you use React.memo everywhere? that clutters the code a bit and I doubt that's the best solution to the performance issues, but we can check this separately

(this is also true elsewhere)